### PR TITLE
Connect buttons with lambda + write to s3 bucket

### PR DESCRIPTION
**Pull request by justineluo**
boba_buddies

### CODE REVIEW

File: lambdaFiles/mainLambda.js [hunk 1]
```diff
@@ -19,28 +23,55 @@ function sendMessage(text, channel) {
...
   console.log(data);
-  switch (data.type) {
-    case "url_verification":
-      verify(data, callback);
+  const resource = data.resource;
+
+  switch (resource) {
+    case "/event-handler":
+      const body = JSON.parse(data.body);
+      //resendText(body["event"], callback);
```
Comment (justineluo) [inline lambdaFiles/mainLambda.js:68]: for some reason this resentText gets call inifinitely ;-; 
need to debug

Comment (justineluo) [reply lambdaFiles/mainLambda.js]: fixed

**Comment by justineluo** [comment]
boba-buddies lambda logs the value of the button pressed: 

Screen Shot 2022-04-06 at 12 55 42 AM

File: lambdaFiles/mainLambda.js [hunk 2]
```diff
@@ -19,28 +29,62 @@ function sendMessage(text, channel) {
...
+    putObjectInS3(text);
+    callback(undefined, responseSuccess);
+  } else {
+    console.log("don't respond to self");
+    callback(undefined, responseSuccess);
   }
-  callback(null);
+}
+
+function handleInteractions(payload, callback) {
```
Comment (VKong6019) [inline lambdaFiles/mainLambda.js:64]: Cracked; should we put the data into s3 from here as well?

File: lambdaFiles/mainLambda.js [hunk 3]
```diff
@@ -19,28 +29,62 @@ function sendMessage(text, channel) {
...
-      resendText(data.event, callback);
+    case "/interactive-handler":
+      const payload = data.body.replace("payload=", "");
+      const decodedPayload = decodeURIComponent(payload).replace(/\+/g, "%20");
+      const payLoadJSON = JSON.parse(decodedPayload);
+      handleInteractions(payLoadJSON, callback);
       break;
     default:
-      callback(null);
+      console.log("What is going on");
```
Comment (VKong6019) [inline lambdaFiles/mainLambda.js:86]: Indeed what is going on

File: lambdaFiles/mainLambda.js (hunk 2 above)
Comment (VKong6019) [reply lambdaFiles/mainLambda.js:64]: On top of that, we'd probably need a if statement when the yes or no button is pressed. Lol if pressed no, then don't do anything?

**Review by VKong6019 — COMMENTED** [review]
Justine cracked

**Review by VKong6019 — APPROVED** [review]
we said to merge this and have the prevent spam button in diff pr

**Pull request merged**